### DISCUSSION TITLE
Postpone workflow until previous workflow completes

### DISCUSSION THREAD

**Discussion by ni-todo-spot** [argoproj/argo-workflows#8955] [Q&A] [answered]
Hey All!

I'm working on a project involving argo as the pipelines framework which executes tasks.

In the project, we got a workflow-template which accepts a `customer-id` parameter.

I was wondering if there is some way to postpone new workflows created by their `customer-id` value?

example:

1. Submit a new workflow with `customer-id=1` (let's call it `a`).
      _Desired result: `a` starts execution immediately._ 

2. Submit a new workflow with `customer-id=2` (let's call it `b`).
      _Desired result: `b` starts execution immediately._  

3.  Submit a new workflow with `customer-id=1` (let's call it `c`) while `a` is still running.
      _Desired result: `c` starts execution immediately._ 

4.  Submit a new workflow with `customer-id=2` (let's call it `d`) while `b` is still running.
      _**Desired result: `d` postponed until `b` is either succeeded or failed/error.**_  

Will be glad to hear your thoughts & comments :)

Thanks in advance!

**Comment by terrytangyuan** [comment, marked as answer]
Use `when` condition to trigger step `d` only when `b` is under certain condition.
